sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ColumnResolutionHelper.scala

```diff
@@ -305,16 +307,16 @@ trait ColumnResolutionHelper extends Logging {
         } else {
           variable
         }
-      }.map(e => Alias(e, nameParts.last)())
+      }.map(e => if (isTopLevel) Alias(e, nameParts.last)() else e )
     }
 
     e.transformWithPruning(_.containsAnyPattern(UNRESOLVED_ATTRIBUTE, TEMP_RESOLVED_COLUMN)) {
```

Comment:
how about skip wrapping `Alias` in the `def resolve`, but check the final value after `e.tranformWithPruning`, if it's not an `Alias`, add one.

Reply:
Removed